app/helpers/compute_resources_vms_helper.rb

```diff
@@ -142,8 +142,12 @@ def show_vm_name?
     controller_name != 'hosts' && controller_name != 'compute_attributes'
   end
 
-  def new_host?(host)
-    host.try(:new_record?)
+  def new_host?(host, options = {})
+    if host.nil? && options[:nil_is_new]
```

Comment:
Sure, I'm not suggesting that `@host` should be removed from the #new or #edit actions, but that it could be removed from #compute_resource_selected action which renders these partials.